tools/build_metal_library.py

```diff
@@ -88,7 +88,9 @@ def Main():
     command += [
       # Embeds both sources and driver options in the output. This aids in
       # debugging but should be removed from release builds.
-      "-frecord-sources",
```

Comment:
Perhaps guard this with an xcode or clang version check?

Reply:
It seems like from Chinmay's comment it's more helpful

Reply:
I'd like for local builds to match CI. CI will be upgrading to Xcode 13 within weeks, so we can move both local and CI builds to `-frecord-sources` at the same time.